app/models/concerns/course/assessment/questions_concern.rb

```diff
@@ -17,7 +17,23 @@ def attempt(submission)
                          map { |answer| [answer.question, answer] }.to_h
 
     map do |question|
-      attempting_answers.fetch(question) { question.attempt(submission) }
+      attempting_answers[question] || question.latest_correct_answer(submission) ||
+        question.attempt(submission)
     end
   end
+
+  # Return the question at the given step.
+  #
+  # @param [Course::Assessment::Submission] submission The submission which contains the answers.
+  # @return [Array<Course::Assessment::Question>] The question at the given the step. The latest
+  #   unfinished question will be returned if the question at the step is not accessible.
+  def step(submission, current_step)
+    # Make sure index is between [0, length - 1]
+    index = [length - 1, [0, current_step.to_i - 1].max].min
+
+    index -= 1 while index > 0 && !fetch(index - 1).finished_by?(submission)
```

Comment:
In v1 there are grading schemas for MCQ. That is, even the student answered correctly, he might not get the full marks(depends on how many times he has tried). 

So we can't tell whether the student answered the question correctly just by their grade. That's why I think there is a need to let concrete questions to override `finished_by?`